AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Core/SQLStatementTests.swift

```diff
@@ -309,6 +310,45 @@ class SQLStatementTests: XCTestCase {
         XCTAssertEqual(variables[1] as? Int, 3)
     }
 
+    /// - Given: a `Model` type
+    /// - When:
+    ///   - the model is of type `Comment`
+    /// - Then:
+    ///   - check if the generated SQL statement is valid
+    ///   - check if an inner join is added referencing `Post`
+    ///   - check if
+    func testSelectStatementFromQueryPredicateWithConnectedField() {
+        let comment = Comment.keys
+
+        let predicate = comment.id == "commentId"
+            && comment.content == "content"
+            && comment.createdAt == nil
+            && comment.post == "PostID"
+
+        let statement = SelectStatement(from: Comment.schema, predicate: predicate)
+        let expectedStatement = """
+        select
+          "root"."id" as "id", "root"."content" as "content", "root"."createdAt" as "createdAt",
+          "root"."commentPostId" as "commentPostId", "post"."id" as "post.id", "post"."content" as "post.content",
+          "post"."createdAt" as "post.createdAt", "post"."draft" as "post.draft", "post"."rating" as "post.rating",
+          "post"."status" as "post.status", "post"."title" as "post.title", "post"."updatedAt" as "post.updatedAt"
```

Comment:
"root"."commentPostId" gets generated by using `field.columnName(forNamespace: rootNamespace) + " as " + field.columnAlias()` which resolves the column name using 
```
extension ModelField: SQLColumn {

    var sqlName: String {
        if case let .belongsTo(_, targetName) = association {
            return targetName ?? name + "Id"
        }
        return name
    }

   func columnName(forNamespace namespace: String? = nil) -> String {
        var column = sqlName.quoted()
        if let namespace = namespace {
            column = namespace.quoted() + "." + column
        }
        return column
    }
}
```

ConditionStatement also generates the columnName differently, using the extension you added on the ModelSchema. Although it would be great if a column name is generated from the same place, but more importantly there seems to be a bug because the select statement generates it from one place while condition statement generates it from another place. if targetName does not exist then it will generate different things. select statement will generate "commentPostId" while the condition statement will fall back to the field "post"